pkg/controllers/management/eks/eks_cluster_handler.go

```diff
@@ -635,3 +651,31 @@ func notFound(err error) bool {
 	}
 	return false
 }
+
+// generateValuesYaml generates a YAML string containing any
+// necessary values to override defaults in values.yaml. If
+// no defaults need to be overwritten, an empty string will
+// be returned.
+func generateValuesYaml() (string, error) {
+	httpProxy := os.Getenv("HTTP_PROXY")
+	httpsProxy := os.Getenv("HTTPS_PROXY")
+	noProxy := os.Getenv("NO_PROXY")
+
+	if httpProxy == "" && httpsProxy == "" && noProxy == "" {
+		// no proxy related environment variables set
+		return "", nil
+	}
+
+	proxySettings := proxyValues{
+		HttpProxy:  httpsProxy,
+		HttpsProxy: httpsProxy,
+		NoProxy:    noProxy,
+	}
+
+	valuesYaml, err := yaml.Marshal(proxySettings)
```

Comment:
If the `proxyValues` struct corresponds to the `values` YAML that we're going to pass in, should `proxyValues` be renamed `environmentOverrides` (or something similar) in case we need to submit non-proxy overrides to the `values.yaml` in the future?

Reply:
I imagine we can just change the name once that is the case, shouldn't effect marshaling at all. I am using valuesYaml everywhere else so it could make sense to change it to be valuesYaml if you are in favor of that.

Reply:
I agree that we could always refactor, but unless there's ever a reason why we would use `proxyValues` directly outside of `generateValuesYaml`, I'd be in favor of just changing the struct name to `valuesYaml` to be more explicit about what it contains / might be updated to contain in the future.

Reply:
okay, fixed the struct to be "eksOperatorValues" and the local variable to be called "values".